rules/do-you-optimize-tinacms-project-for-clarity-performance-and-reliable-builds/rule.md

```diff
@@ -0,0 +1,109 @@
+  * This approach ensures efficient data retrieval and reduces the server load at build time.
+
+## 2. Improving Runtime Performance
+
+Optimizing runtime performance is key to delivering a fast and responsive user experience.
+
+### Bad Practices:
+
+* **Using Client-Side Requests Instead of Cached Values from Server-Side Components:**
+
+  * This approach bypasses pre-fetched, cached data, increasing server load and slowing down the application.
+  * It negates the benefits of static site generation, where data should be ready and waiting.
+
+### Common Practices:
+
+* **Utilizing Next.js caching via fetch:** 
+
+  * Next.js’s fetch API is designed to cache by default during static generation and server-side rendering.
+
+### Good Practices:
+
+* **Utilizing Caching at the Highest Level of the Application:**
+
+  * By caching data at the top level, you ensure that it is readily available throughout your application, reducing redundant data fetching and improving overall performance.
+  * This approach allows for more efficient data management and quicker response times.
+* **Using Dedicated Caching Packages:**
+
+  * [**Redis**:](https://redis.io/fr/solutions/cas-dutilisation/cache/) An in-memory data structure store, Redis is fast and versatile, making it ideal for caching data in memory. 
+  * **[Node-cache](https://www.npmjs.com/package/node-cache)**: A lightweight, in-memory caching solution for Node.js applications.
+
+### Gold Plating:
+
+* **Implementing a Singleton Service for Data Requests:**
+
+  * A singleton service that encapsulates general requests and provides data via cache ensures consistent data retrieval and minimizes repetitive API calls, thereby enhancing performance.
+
+## 3. Preventing Build Crashes
+
+To ensure smooth and reliable builds, it’s important to follow best practices that prevent excessive server load and manage data efficiently.
+
+### Best Practices:
+
+* **Avoid Isolating Tina Client Calls in Multiple Subcomponents:**
+
+  * Centralize data fetching to avoid overwhelming the build process with excessive API requests.
+* **Cache Large Datasets at a Top Level:**
+
+  * By caching large datasets at a top level, you reduce the number of data fetches during the build process, lowering the risk of overwhelming the system.
+* **Avoid Using Deeply Nested References:**
+
+  * Simplifying your data structure and avoiding deeply nested references can help ensure smoother and faster builds.
+* **Write custom Tina queries:**
+
+  * You can improve build generation and prevent build crash by creating your own Tina GraphQL queries.
+
+### One last option:
+
+* If a build fails due to overload activities, you can consider reducing the server load by modifying the `next.config.js` file as follows, but this should be done cautiously. This will slow the build process.
```

Comment:
this doesn't sound great :O Is this a problem with Tina Cloud?

Reply:
Given that the starvation issue has been fixed in TinaCloud - this isn't an issue anymore

Reply:
✅ Done, I deleted that part.